nuget/lib/dependabot/nuget/native_discovery/native_discovery_json_reader.rb

```diff
@@ -41,6 +41,17 @@ def self.testonly_clear_discovery_files
         FileUtils.rm_rf(discovery_directory)
       end
 
+      sig { params(error_if_missing: T::Boolean).void }
+      def self.debug_report_discovery_files(error_if_missing:)
+        if File.exist?(discovery_map_file_path)
+          Dependabot.logger.info("Discovery map file (#{discovery_map_file_path}) contents: " \
+                                 "#{File.read(discovery_map_file_path)}")
+          Dependabot.logger.info("Discovery files: #{Dir.glob(File.join(discovery_directory, '*'))}")
+        elsif error_if_missing
+          Dependabot.logger.error("discovery map file missing")
```

Comment:
This allows PR creation to continue, but will make a handy search point in the logs.